### ISSUE TITLE
2 AM Ramble About Custom GregTech Radiation Mechanic

### ISSUE THREAD

**Issue by swagXDragonSlayer46YT** [GregTechCEu/gt-ideas#27]
## Details
Since it seems that GregTech is implementing heavier elements that are highly toxic and radioactive, I guess it would be reasonable to add in new gameplay mechanics that require the player to use extra care when handling stuff like polonium, astatine, etc.

There are many types of radiation in real life ranging from particle radiation to electromagnetic radiation. This radiation will not only hurt players but will also damage electronics and could make machines unusable.

Alpha particles (emitted by unstable nuclei that undergo alpha decay) - Easiest radiation to block, even paper can protect against this.

β- Decay (high energy electrons) - Also easy to block, but it is better to use materials with low atomic weight such as water, plastic, or wood, to prevent making more radiation due to Bremsstrahlung X-rays. β+ Decay emits positrons instead, which produce Gamma Rays when they collide with electrons in normal matter.

X-Ray or Gamma Rays - Harder to block. Almost any material can be used but it has to be thick. Lead is the one of the most common materials used to block this radiation. Nuclear Reactors use specialized types of concrete as well for extra shielding. This type of radiation has side effects on living beings and electronics due to its ability to penetrate materials and affect atoms in them.

"Most nuclear reactors use thick concrete shields to create a bioshield with a thin water-cooled layer of lead on the inside to protect the porous concrete from the coolant inside. The concrete is also made with heavy aggregates, such as [Baryte](https://en.wikipedia.org/wiki/Baryte) or MagnaDense ([Magnetite](https://en.wikipedia.org/wiki/Magnetite)), to aid in the shielding properties of the concrete." - https://en.wikipedia.org/wiki/Radiation_protection#Radiation_shielding

Finally there is neutron radiation. This type of radiation is the most damaging. This radiation can be mitigated by using neutron reflectors (graphite, beryllium, tungsten carbide, etc) or absorbed by light materials such as water or polyethylene. It also degrades reactor materials over time.

"Graphite [neutron moderator](https://en.wikipedia.org/wiki/Neutron_moderator) blocks are especially susceptible to this effect, known as [Wigner effect](https://en.wikipedia.org/wiki/Wigner_effect), and must be annealed periodically. The [Windscale fire](https://en.wikipedia.org/wiki/Windscale_fire) was caused by a mishap during such an annealing operation." - https://en.wikipedia.org/wiki/Neutron_radiation

Radioactive materials, stored in inventories or chests or Quantum Storages or whatever, would produce one or more of the four types of radiation listed above based on how they decay. The same goes for radioactive liquids/gases placed in the world or stored in fluid tanks/buses or whatever is capable of storing liquids/gases. Radioactive machine blocks or multiblocks (such as fission reactors) will also produce one or more of these types of radiation. Essentially there will be 4 different types of maps corresponding to each type of radiation, which will each have their own effects on blocks and entities in those maps.

Each individual chunk will have its own counter for the four types of radiation above. Individual blocks will not have their own radiation because not even censoredasm will save us from the supreme lag that will create. Particles from alpha decay and beta decay will be ez to take care of. However, Gamma radiation and neutron radiation will spread very far to other chunks if not taken care of.

But how exactly will radiation be taken care of? The way radiation will be blocked might be that radioactive blocks will contribute less radiation to the chunk if the right material/block is used to cover them on their sides (with all sides being the best fpr blocking). For example, a block producing neutron radiation will not increase radiation in the chunk if it is covered by a lot of water around it. The more water, the less radiation (neutron radiation is annoying and does not go away that easily).

Increasing gamma radiation and neutron radiation in chunks will lead to increased chances of negative effects on entities and machines in the chunks. For example, machines might stop working. You might get a negative status effect. Or if you want to go GTNH mode just make everything explode when the slightest thing goes wrong.

However, radiation can sometimes be useful for stuff related to biology and genetics (similar to how radioactive materials were sometimes used to assist in bacterial vat recipes in GT:NH) But we will not get into that for now.

New machine covers could be added that provide radiation protection for machines. Inversely, new machine covers could be added that prevents machines from contributing certain types of radiation to chunks.

Hazmat suits and geiger counters could also be added for dealing with radiation.

By making radioactive materials annoying to deal with, it could also lead to another interesting gameplay choice. Long term storage and disposal of radioactive materials. Imagine gregtech bases with spent nuclear waste bunkers hundreds of blocks from the main area. That will be pretty "Greggy" as Regian would have put it.

## Sources
Broader article on the effects of radiation on materials and human beings: https://en.wikipedia.org/wiki/Radiation_damage

**Comment by tolyabi228** [comment]
to make the waste storing part to work methods of simple deletion of radioactive materials by lava/cactus/else should be somehow punishing, lets say add total radiation from deleted material, *10 it and release into environment/chunks.
and i guess some isotopes should have functionality of being used for radioisotopic batteries(akin to how its done in HBMs HTM but not nested and way lower output), they may be similar to old ic2 batteries that were crafted fully charged, but are able to slowly recharge themselves for some time, after expiring you treat them like rest of waste

**Comment by Zalgo239** [comment]
I think it's worth making a distinction between radiation and radioactive contamination, the total radiation in a chunk should be the sum of the dose given by sources like inventories and machines and the presence of radioactive particles in the air/ground/water/whatever.

The first is immediately negated by shielding machines and inventories. 
The second is caused by radioactive items or fluids being thrown on the ground or despawning, it lingers around and requires cleanup operations to get rid of (think of the scrubbers from nuclearcraft), while also being able to spread to other chunks.

**Comment by swagXDragonSlayer46YT** [comment]
> I think it's worth making a distinction between radiation and radioactive contamination, the total radiation in a chunk should be the sum of the dose given by sources like inventories and machines and the presence of radioactive particles in the air/ground/water/whatever.
> 
> The first is immediately negated by shielding machines and inventories. The second is caused by radioactive items or fluids being thrown on the ground or despawning, it lingers around and requires cleanup operations to get rid of (think of the scrubbers from nuclearcraft), while also being able to spread to other chunks.

I agree with this idea. Furthermore, neutron radiation (which will fall into the first category) will cause radioactive contamination due to possible neutron activation.

**Comment by Gliese-832-c** [comment]
Igblon, would it be possible for you to reopen some sort of alt account on Discord? I understand you want to focus on your channel more, but using an alt would allow you to avoid distractions and only join relevant servers (basically some sort of communication channel with us). It would be nice if you could do that so we can discuss various ideas. There's one in particular we'd like your input on, but it's classified until we have a working prototype ready to be released on CurseForge, and GitHub doesn't have any sort of DM features.

**Comment by Igblon** [comment]
> Igblon, would it be possible for you to reopen some sort of alt account on Discord? I understand you want to focus on your channel more, but using an alt would allow you to avoid distractions and only join relevant servers (basically some sort of communication channel with us). It would be nice if you could do that so we can discuss various ideas. There's one in particular we'd like your input on, but it's classified until we have a working prototype ready to be released on CurseForge, and GitHub doesn't have any sort of DM features.

I would rather they send me an email. igblonbusiness@outlook.com. Response time will be < 24 Hours. Don't worry about conversations taking too long over email. Just put all of your questions in one email and I will make a detailed response.
